test.js

```diff
@@ -91,3 +92,40 @@ test('kill from port', async t => {
 test('error when process is not found', async t => {
 	await t.throwsAsync(fkill(['notFoundProcess']), /Killing process notFoundProcess failed: Process doesn't exist/);
 });
+
+function testKillDescendant(t, name = '') {
+	const fixture = path.resolve('fixtures', 'descendant');
+	const cp = childProcess.spawn('node', [fixture, name]);
+
+	cp.stdout.setEncoding('utf8');
+	cp.stdout.on('data', async chunk => {
+		const descendantPid = parseInt(chunk, 10);
+		t.is(typeof descendantPid, 'number');
+
+		const opts = {
+			tree: true,
+			force: process.platform === 'win32'
+		};
+
+		if (name) {
+			await fkill(name, opts);
+		} else {
+			await fkill(cp.pid, opts);
+		}
+
+		// Ensure all the processes has time to exit
+		await delay(400);
+
+		t.false(await processExists(cp.pid));
+		t.false(await processExists(descendantPid));
+		t.end();
+	});
+}
+
+// eslint-disable-next-line ava/test-ended
+test.cb('kill all descendants tree by pid', testKillDescendant);
+
+if (process.platform !== 'win32') {
```

Comment:
Why is this not tested on Windows?

Reply:
To test killing a tree of processes I need a parent process that can spawn a descendant.
To test killing a process by its name I need a fixture that can change name, otherwise I'm not able to pinpoint that process easily.
The fixture `descendant` that I'm using can change its own name via `process.title` but cannot on Windows. I've just tested this on Windows 10.

There are two solutions that comes to mind:
- create a binary fixture for Windows only that spawn a descendant and has a custom process name
- kill indiscriminately all processes called `node.exe`
- change the process title using some other tool, better if built-in in Windows (but I've found nothing so far and haven't searched enough)